cmd/natc/natc.go

```diff
@@ -192,6 +214,21 @@ func (c *connector) serveDNS() {
 	}
 }
 
+func lookupDestinationIP(domain string) ([]netip.Addr, error) {
+	netIPs, err := net.LookupIP(domain)
```

Comment:
consider `net.Resolver`'s `LookupIP` and you can specify `ip4` vs `ip6` to only get records of the address type corresponding to the incoming DNS response.